### DISCUSSION TITLE
support UnifiedPush notifications

### DISCUSSION THREAD

**Discussion by flexagoon** [Docile-Alligator/Infinity-For-Reddit#957] [Ideas]
[UnifiedPush](https://unifiedpush.org) is a FOSS selfhostable push notification server specification which makes it easy for apps to send live notifications without Google Cloud Messaging. Perhaps you can add support for it in Infinity? It'll allow for smaller notification checking interval while improving the battery life of the device. You'll still be able to leave the default polling-based notification approach as a fallback in case the user doesn't have UP set up on their device.

**Comment by MrDrMcCoy** [comment]
As much as I would love to see something like this, support must first come from either Reddit or by someone else making a server that polls Reddit and then relays any notifications via UnifiedPush. Simply implementing UnifiedPush support in Infinity would be useless if there's nothing that will ever send a notification that way.

**Comment by karmanyaahm** [comment]
Not sure if Reddit allows all servers (those not from Google, Mozilla, Apple, etc), but Reddit does have WebPush. WebPush can send notifications to arbitrary (including UnifiedPush) URLs, that's how Mastodon works with UP.

WebPush registration API call in a browser:
![image](https://user-images.githubusercontent.com/32671690/209590936-b2416eee-3c72-4f2e-9b5f-596cf6c24316.png)
![image](https://user-images.githubusercontent.com/32671690/209591025-532ab966-cabb-4a27-a33a-ace12a58efa2.png)

Of course, someone will have to reverse engineering Reddit's pusher a bit.

**Comment by Sayasam42** [comment]
I fully agree that, one way or another, push notifications would be a great addition to this wonderful app.
